### PR TITLE
Initial2

### PR DESCRIPTION
**Pull request by chris-giblin**
This PR consolidates changes made during the discussion in PR https://github.com/strimzi/topic-encryption/pull/9 which this PR supersedes

### CODE REVIEW

File: doc/getting-started.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
+# Getting started
+
+Requirements:
+- a Kafka instance, version 2.8.0 or older, which you can configure
```
Comment (tombentley) [inline doc/getting-started.md:4]: Where does the 2.8.0 requirement come from? Or is this really a statement about what you've tested?

File: encmod/pom.xml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        <artifactId>log4j-api</artifactId>
+    </dependency>
+    <dependency>
+        <groupId>org.apache.logging.log4j</groupId>
+        <artifactId>log4j-core</artifactId>
+    </dependency>
+    <dependency>
+        <groupId>org.apache.logging.log4j</groupId>
+        <artifactId>log4j-slf4j-impl</artifactId>
+    </dependency>
```
Comment (tombentley) [inline encmod/pom.xml:29]: I guess these can be `test` scoped in this module, and then let the `vertx-proxy` module pull them in as runtime dependencies.

File: vertx-proxy/pom.xml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+	 <artifactId>vertx-config</artifactId>
+	</dependency>
+    <dependency>
+	  <groupId>org.apache.kafka</groupId>
+	  <artifactId>kafka-clients</artifactId>
+    </dependency>
+	<dependency>
+	  <groupId>io.strimzi</groupId>
+	  <artifactId>encmod</artifactId>
+	  <version>0.0.1-SNAPSHOT</version>
```
Comment (tombentley) [inline vertx-proxy/pom.xml:41]: This can be `${project.version}`?

**Review by tombentley — APPROVED** [review]
A few minor points, then let's get this merged!

File: vertx-proxy/pom.xml (hunk 3 above)
Comment (chris-giblin) [reply vertx-proxy/pom.xml]: Indeed. I just committed it. Thanks.

File: doc/getting-started.md (hunk 1 above)
Comment (chris-giblin) [reply doc/getting-started.md:4]: [Kafka 3.1.0](https://developers.redhat.com/articles/2022/02/10/kafka-monthly-digest-january-2022#kafka_3_1_0_release) introduces topic id's to FETCH requests ([KIP 516](https://cwiki.apache.org/confluence/display/KAFKA/KIP-516%3A+Topic+Identifiers)).  This version of topic encryption does not support topic id's, rather it performs string-matching to identify topic partitions in protocol messages.
Actually Kafka 3.0 will also work with this release, but 2.8.0 is what is tested.

Comment (chris-giblin) [reply doc/getting-started.md:4]: I have created an issue to address topic IDs: https://github.com/strimzi/topic-encryption/issues/17

File: encmod/pom.xml (hunk 2 above)
Comment (chris-giblin) [reply encmod/pom.xml]: Done

File: doc/getting-started.md (hunk 1 above)
Comment (tombentley) [reply doc/getting-started.md:4]: Thanks!

**Pull request merged**